.vscode/tasks.json

```diff
@@ -0,0 +1,38 @@
+{
+    // See https://go.microsoft.com/fwlink/?LinkId=733558
+    // for the documentation about the tasks.json format
+    "version": "2.0.0",
+    "tasks": [
+        {
+            
+            "label": "Create venv with all exporters",
+            "type": "shell",
+            "linux": {
+                "command": "${workspaceFolder}/tools/prepare_virtual_env.sh ; ${workspaceFolder}/venv/Scripts/activate ; pip install -r requirements.txt",
+            },
+            "windows": {
+                "command": "@()|${workspaceFolder}/tools/prepare_virtual_env.bat ; ${workspaceFolder}/venv/Scripts/Activate.ps1 ; pip install -r requirements.txt",
```

Comment:
Not sure I like using these scripts for vscode.

Reply:
What's the alternative besides "Call exactly this manually"?

Reply:
The script is just a fancy eyelet with lots of checks around creating a venv and installing requirements.txt

Doesn't the python plugin create venvs automatically?

Reply:
Not automatically but you can create it from there via `Python: Create environment`
But that would miss all checks for the correct python version, would ask what you want to install from the pyproject.toml (exporters, gui, server, test, typing) and would also miss the installation of the pre-commit.

Reply:
ping

Reply:
IMO if you're installing the entire requirements.txt, you'd also skip the python version check. My view is that someone managing the venv via VSCode is now more technical and we shouldn't restrict their choices,

Reply:
There is nothing forcing you to use the tasks defined in this file, it's just more of a convenient thing if you have no idea what to do.

Hmmm but it's executed as one thing, so you can't skip the check with this task.

So... you are suggesting to remove the task and write in the readme how to create the venv within  vs code manually?

Reply:
ping

Reply:
I am not sure the best things to choose. The intent of these scripts are to provide an easy way for less technical people to run Randovania, mostly focused on these who just want to edit the database.

The script clearly isn't what you're looking for because you immediately install more packages after.

Providing some documentation (and links to where read more?) about how to manage the venv might be best? Understanding what's going on is essential if you want to also go a bit deeper, such as also having the patcher installed from a git clone.

Reply:
IMO installing the exporter requirements makes things more user friendly. It avoids the following situation with:

"I cant export!"
\- "install the requirements, see the readme for details"
"What does the readme mean i have to see the tools\start_client script for info?"
....

Having a "one click button" is much nicer for that. It not like it comes with that many negatives: It works for all desktop OS, and you only have to download a few hundred MB more, which by today's standards isn't really that big of a deal.